src/lib.rs

```diff
@@ -9,12 +9,23 @@ pub mod metrics;
 pub mod wrapper;
 use crate::bloom::command_handler;
 use crate::bloom::data_type::BLOOM_TYPE;
+use crate::bloom::utils::valid_server_version;
 use valkey_module_macros::info_command_handler;
 
 pub const MODULE_NAME: &str = "bf";
 
-fn initialize(_ctx: &Context, _args: &[ValkeyString]) -> Status {
-    Status::Ok
+fn initialize(ctx: &Context, _args: &[ValkeyString]) -> Status {
+    let ver = ctx
+        .get_redis_version()
+        .expect("Unable to get server version!");
+    if !valid_server_version(ver) {
+        ctx.log_warning(
+            "Current server version doesn't satisfy minimal required version for bloom module!",
+        );
```

Comment:
1. We can use error level logging since this is a critical error
2. We can make the log more informative with the exact version. You can replace X with the exact version


```suggestion
        ctx.log_error(
            "The minimum supported Valkey server version for bloom module is {}",
            X
        );
```

Reply:
There has no Error level in the valkeymodule-rs, maybe `log_io_error` is what you referring to?